bazel_tools/haskell-c2hs-windows.patch

```diff
@@ -0,0 +1,87 @@
+diff --git a/haskell/c2hs.bzl b/haskell/c2hs.bzl
+index f4f9f42..f45bc21 100644
+--- a/haskell/c2hs.bzl
++++ b/haskell/c2hs.bzl
+@@ -22,6 +22,7 @@ def _c2hs_library_impl(ctx):
+     cc = cc_interop_info(ctx)
+     args = hs.actions.args()
+     c2hs = ctx.toolchains["@rules_haskell//haskell/c2hs:toolchain"].c2hs
++    c2hs_exe = ctx.toolchains["@rules_haskell//haskell/c2hs:toolchain"].c2hs_exe
+ 
+     if len(ctx.files.srcs) != 1:
+         fail("srcs field should contain exactly one file.")
+@@ -39,6 +40,7 @@ def _c2hs_library_impl(ctx):
+     args.add("-C-includeghcversion.h")
+     args.add_all(["-C" + x for x in cc.cpp_flags])
+     args.add_all(["-C" + x for x in cc.include_args])
++    args.add_all(ctx.attr.extra_args)
+ 
+     dep_chi_files = [
+         dep[C2hsLibraryInfo].chi_file
+@@ -59,22 +61,36 @@ def _c2hs_library_impl(ctx):
+         (version_macro_headers, version_macro_flags) = version_macro_includes(dep_info)
+         args.add_all(["-C" + x for x in version_macro_flags])
+ 
++    (inputs, input_manifests) = ctx.resolve_tools(tools = [c2hs])
```

Comment:
@aherrmann-da I need `c2hs` to be a target for this to work but I couldn’t figure out how I could then get the path from the target so as a hacky workaround I added `c2hs_exe` which points to the file and `c2hs` which points to the target to the toolchain. What’s the non-hacky way to accomplish this?

Reply:
Yes, the information that it's a single executable file is unfortunately lost once you only have the `Target`. The separate `c2hs_exe` seems like a good solution. I've essentially done the same [with the cc_wrapper](https://github.com/tweag/rules_haskell/pull/1039/files#diff-bd788ee16d6ba21ea60c42efea01aba3R163-R166).